packages/loaders/src/Dots.js

```diff
@@ -5,112 +5,86 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import React from 'react';
+import React, { useEffect, useRef } from 'react';
 import PropTypes from 'prop-types';
 
-import { retrieveXCoordinate, retrieveYCoordinate, KEYFRAME_MAX } from './utils/dot-coordinates';
-import { DotsCircle, StyledSVG } from './styled-elements';
-import ScheduleContainer from './containers/ScheduleContainer';
+import {
+  DotsOneCircle,
+  DotsTwoCircle,
+  DotsThreeCircle,
+  StyledSVG,
+  LoadingPlaceholder
+} from './styled-elements';
+import { useSchedule } from '@zendeskgarden/container-schedule';
 
 const COMPONENT_ID = 'loaders.dots';
 
-export default class Dots extends React.Component {
-  static propTypes = {
-    /**
-     * Size of the loader. Can inherit from `font-size` styling.
-     **/
-    size: PropTypes.any,
-    /**
-     * Velocity (speed) of the animation. Between -1 and 1.
-     * This should only be maniuplated at extreme sizes.
-     **/
-    velocity: PropTypes.number,
-    /**
-     * Color of the loader. Can inherit from `color` styling.
-     **/
-    color: PropTypes.string,
-    /**
-     * Delay in MS to begin loader rendering. This helps prevent
-     * quick flashes of the loader during normal loading times.
-     **/
-    delayMS: PropTypes.number
-  };
+/** @component */
+export default function Dots({
+  size = 'inherit',
+  color = 'inherit',
+  duration = 1250,
+  delayMS = 750,
+  ...other
+}) {
+  const { delayComplete } = useSchedule({ duration, delayMS });
+  const dotOne = useRef(null);
+  const dotTwo = useRef(null);
+  const dotThree = useRef(null);
 
-  static defaultProps = {
-    size: 'inherit',
-    color: 'inherit',
-    velocity: 0.05,
-    delayMS: 750
-  };
+  useEffect(() => {
+    // gah user agent sniffing! This is so we only run the effect in IE11
+    if (/MSIE \d|Trident.*rv:/u.test(navigator.userAgent) && delayComplete) {
```

Comment:
Before you banish me for user agent sniffing here me out, we can support CSS based animations in all browsers except IE11 *but* we can query the computed value of the animation in IE11 and update the transform attribute to enable the animation to work just fine.

This simplifies the code greatly and also allows us to remove the `velocity` prop in favour of the `duration` prop without loss of perf or fps.

If you have a nicer way of detecting svg animation support let me know! I tried listening to `animationiteration` event on mount but that made for a weird first loop in the good browsers where it would do the CSS animation and the attribute transform.

Reply:
@ryanseddon is there something that could be cribbed from this? https://github.com/Modernizr/Modernizr/blob/master/feature-detects/svg/smil.js. i.e. `/SVGAnimate/.test(({}).toString.call(document.createElementNS('http://www.w3.org/2000/svg', 'animate')))`

Reply:
Wasn't like I was a core contributor to that library or anything! Good idea I'll test this and see if we're good.